### PR TITLE
Allow adding custom classes to header, content, footer, etc.

### LINKED ISSUE #1440: Add ability to set custom classes for different parts (container, header, content, footer, etc.)
## New feature motivation

There's often a need to apply custom styles to these blocks:

- container (`.swal2-container`), there was `customContainerClass` previously, deprecate it
- header (`.swal2-header`)
- title (`.swal2-title`)
- close button (`.swal2-close`)
- content (`.swal2-content`)
- input (`.swal2-input`), there was `inputClass` previously, deprecate it
- actions (`.swal2-actions`)
- footer (`.swal2-footer`)

E.g. in this example of side-panel: https://sweetalert2.github.io/sweetalert2-examples/misc/side-panel/ we're using this CSS:

```css
.swal2-container {
  padding: 0;
}

.swal2-popup {
  border-radius: 0;
}

.swal2-content {
  flex: 1;
}
```

Which is wrong because `swal2-*` classnames are not the public API and might be changed at any time (the possibility of that is very small, but still).

## New feature implementation

Add the ability to pass an object as `customClass`, e.g. 

```js
Swal.fire({
  ...
  customClass: {
    container: 'container-class',
    popup: 'popup-class',
    header: 'header-class',
    title: 'title-class',
    closeButton: 'close-button-class',
    content: 'content-class',
    input: 'input-class',
    actions: 'actions-class',
    footer: 'footer-class'
  }
  ...
}
```

Also, deprecate `customContainerClass`, `confirmButtonClass`, `cancelButtonClass`, `imageClass`, `inputClass`

### PR DESCRIPTION
**Pull request by limonte**
Closes #1440 

```js
Swal.fire({
  ...
  customClass: {
    container: 'container-class',
    popup: 'popup-class',
    header: 'header-class',
    title: 'header-class',
    closeButton: 'close-button-class',
    content: 'content-class',
    input: 'input-class',
    actions: 'actions-class',
    footer: 'footer-class'
  }
  ...
}
```

(API docs https://github.com/sweetalert2/sweetalert2.github.io/issues/65)

---

Also, `customContainerClass`, `confirmButtonClass`, `cancelButtonClass`, `imageClass`, `inputClass` are deprecated (https://github.com/sweetalert2/sweetalert2.github.io/issues/66)

---

Alos, expose `Swal.getHeader()` (API docs https://github.com/sweetalert2/sweetalert2.github.io/issues/64)

---

Also, in scope of this PR there's a small improvement of the deprecation warning:

Before:

image

After:

image

### CODE REVIEW

**Comment by limonte** [comment]
@gverni  the PR is ready for review 🚀

**Comment by gverni** [comment]
Thnaks @limonte, i'm on it...

File: src/instanceMethods/_main.js [hunk 1]
```diff
@@ -377,6 +377,9 @@ export function _main (userParams) {
       if (innerParams.inputClass) {
         dom.addClass(inputContainer, innerParams.inputClass)
       }
+      if (innerParams.customClass) {
```
Comment (gverni) [inline src/instanceMethods/_main.js]: For **performance reasons**, this should be: 

```javascript
if (innerParams.customClass.input) 
```

If you set a breakpoint inside the `if` you will see that is called 5 times (one for each input types) **even if `customClass.input` it's undefined**. I think we should avoid that...

File: sweetalert2.d.ts [hunk 2]
```diff
@@ -452,12 +464,30 @@ declare module 'sweetalert2' {
 
         /**
          * A custom CSS class for the modal.
+         * If a string value is provided, the classname will be applied to the popup.
```
Comment (gverni) [inline sweetalert2.d.ts]: It works also with an array of strings. Do you want to add that?

File: src/instanceMethods/_main.js (hunk 1 above)
Comment (limonte) [reply src/instanceMethods/_main.js]: the performance difference is almost nothing here, what you're suggesting will cause this edge case to fail:

image

File: sweetalert2.d.ts (hunk 2 above)
Comment (limonte) [reply sweetalert2.d.ts]: Done.

**Comment by limonte** [comment]
> Also, what's the rationale behind leaving confirmButtonClass, cancelButtonClass, imageClass out of customClass? Shouldn't they be added as well?

Good catch! Done.

File: sweetalert2.d.ts [hunk 3]
```diff
@@ -452,12 +467,33 @@ declare module 'sweetalert2' {
 
         /**
          * A custom CSS class for the modal.
+         * If a string (or an array of strings) value is provided, the classname will be applied to the popup.
```
Comment (gverni) [inline sweetalert2.d.ts]: Sorry @limonte, actually the array of strings change was meant for the values of the keys inside the object. So sorry about asking you to change it back again.

Comment (limonte) [reply sweetalert2.d.ts]: No problem! Reverted.

**Comment by gverni** [comment]
Thanks @limonte, for me it's good now. I'll create some more example at https://github.com/sweetalert2/sweetalert2-examples and update the side bar one.

**Pull request merged**